EIPS/eip-6682.md

```diff
@@ -0,0 +1,161 @@
+---
+eip: 6682
+title: NFT Flashloans
+description: Minimal interface for ERC-721 NFT flashloans
+author: out.eth (@outdoteth)
+discussions-to: https://ethereum-magicians.org/t/eip-6682-nft-flashloans/13294
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-03-12
+requires: 721, 3156
+---
+
+## Abstract
+
+This standard is an extension of the existing flashloan standard ([ERC-3156](./eip-3156.md)) to support [ERC-721](./eip-721.md) NFT flashloans. It proposes a way for flashloan providers to lend NFTs to contracts, with the condition that the loan is repaid in the same transaction along with some fee.
+
+## Motivation
+
+The current flashloan standard, [ERC-3156](./eip-3156.md), only supports [ERC-20](./eip-20.md) tokens. ERC-721 tokens are sufficiently different from ERC-20 tokens that they require an extension of this existing standard to support them. 
+
+In most cases, the handling of fee payments will be desired to be paid in a seperate currency to the loaned NFTs because NFTs themselves cannot always be fractionalized. Consider the following example where the flashloan provider charges a 0.1 ETH fee on each NFT that is flashloaned; The interface must provide methods that allow the borrower to determine the fee rate on each NFT and also the currency that the fee should be paid in.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Contract Interface
+
+```solidity
+pragma solidity ^0.8.19;
+
+interface IERC6682 {
+    /// @dev The address of the token used to pay flash loan fees.
+    function flashFeeToken() external view returns (address);
+
+    /// @dev Whether or not the NFT is available for a flash loan.
+    /// @param token The address of the NFT contract.
+    /// @param tokenId The ID of the NFT.
+    function availableForFlashLoan(address token, uint256 tokenId) external view returns (bool);
+}
+```
+
+The `flashFeeToken` function MUST return the address of the token used to pay flash loan fees.
+
+If the token used to pay the flash loan fees is ETH then `flashFeeToken` MUST return `address(0)`.
+
+The `availableForFlashLoan` function MUST return whether or not the `tokenId` of `token` is available for a flashloan. If the `tokenId` is not currently available for a flashloan `availableForFlashLoan` MUST return `false` instead of reverting.
+
```

Comment:
I'd like to see a little note saying implementers are supposed to implement `IERC3156FlashLender`. It isn't obvious from the specification text how you're actually supposed to borrow the token.

Reply:
updated